### ISSUE TITLE
Log error & traffic loss when resetting 2 devices, in a 3 devices setup.

### ISSUE THREAD

**Issue by rozmanro1** [acassen/keepalived#1573]
Hi all,
Would like your advice regarding the next issue:

I have a 3 devices setup, all 3 routers connected to a SWITCH (8 ports).
Device_A: 192.168.20.1 on eth4 -> connected to switch port eth2 using vlan 100.
Device_B: 192.168.20.2 on eth4 -> connected to switch port eth3 using vlan 100.
Device_C: 192.168.20.4 on eth4 -> connected to switch port eth4 using vlan 100.

Virtual_IP: 192.168.20.3

**Configuration file:**
conf file is the same on each one of them,only the priority is different:
A - vrid50, priority 200
B - vrid50, priority 100
C - vrid50, priority 80

# cat /etc/keepalived/keepalived.conf
! Configuration File for keepalived

global_defs {
        vrrp_version 3
}

vrrp_instance VI_1 {
    interface eth4.100
    virtual_router_id 50
    priority 200
    advert_int 1
    preempt
    use_vmac

    virtual_ipaddress {
        192.168.20.3
    }
}#
#

A is the master, B & C are backups, traffic is running ok and all good.

**The issue:**
When we reset devices B & C, device A is logging some error messages and the traffic is lost.
The traffic is back on only after 2 min after devices B & C are back up.

**This is not happening in a 2 device setup.**
**This is not happening if we reset only one between B & C devices.**

**Keepalived version**
Keepalived v2.0.15 (03/11,2020) with custom linux kernel.

**Keepalived Logs from device A**
May 12 14:18:43 Keepalived[18762]: Starting Keepalived v2.0.15 (04/26,2020), git commit BuildRoot_2.3_GA.007-1-g914a74f+
May 12 14:18:43 Keepalived[18762]: Running on Linux 4.19.19 #1 SMP Sun Apr 26 15:02:08 IDT 2020 (built for Linux 4.19.19)
May 12 14:18:43 Keepalived[18762]: Command line: 'keepalived' '-D' '--log-facility=0'
May 12 14:18:43 Keepalived[18762]: Opening file '/etc/keepalived/keepalived.conf'.
May 12 14:18:43 Keepalived[18762]: Remove a zombie pid file /var/run/keepalived.pid
May 12 14:18:43 Keepalived[18762]: Remove a zombie pid file /var/run/vrrp.pid
May 12 14:18:43 Keepalived[18763]: Starting VRRP child process, pid=18764
May 12 14:18:43 Keepalived_vrrp[18764]: Registering Kernel netlink reflector
May 12 14:18:43 Keepalived_vrrp[18764]: Registering Kernel netlink command channel
May 12 14:18:43 Keepalived_vrrp[18764]: Opening file '/etc/keepalived/keepalived.conf'.
May 12 14:18:43 Keepalived_vrrp[18764]: (VI_1) Found matching interface vrrp.50
May 12 14:18:43 Keepalived_vrrp[18764]: Assigned address 192.168.20.1 for interface eth4.100
May 12 14:18:43 Keepalived_vrrp[18764]: Registering gratuitous ARP shared channel
May 12 14:18:43 Keepalived_vrrp[18764]: (VI_1) removing VIPs.
May 12 14:18:43 Keepalived_vrrp[18764]: (VI_1) Entering BACKUP STATE (init)
May 12 14:18:43 Keepalived_vrrp[18764]: VRRP sockpool: [ifindex(30), family(IPv4), proto(112), unicast(0), fd(16,17)]
May 12 14:18:43 Keepalived_vrrp[18764]: (VI_1) received lower priority (100) advert from 192.168.20.2 - discarding
May 12 14:18:44 Keepalived_vrrp[18764]: (VI_1) received lower priority (100) advert from 192.168.20.2 - discarding
May 12 14:18:45 Keepalived_vrrp[18764]: (VI_1) received lower priority (100) advert from 192.168.20.2 - discarding
May 12 14:18:46 Keepalived_vrrp[18764]: (VI_1) Receive advertisement timeout
May 12 14:18:46 Keepalived_vrrp[18764]: (VI_1) Entering MASTER STATE
May 12 14:18:46 Keepalived_vrrp[18764]: (VI_1) using locally configured advertisement interval (1000 milli-sec)
May 12 14:18:46 Keepalived_vrrp[18764]: (VI_1) setting VIPs.
May 12 14:18:46 Keepalived_vrrp[18764]: Sending gratuitous ARP on vrrp.50 for 192.168.20.3
May 12 14:18:46 Keepalived_vrrp[18764]: (VI_1) Sending/queueing gratuitous ARPs on vrrp.50 for 192.168.20.3
May 12 14:18:46 Keepalived_vrrp[18764]: Sending gratuitous ARP on vrrp.50 for 192.168.20.3
May 12 14:18:46 Keepalived_vrrp[18764]: Sending gratuitous ARP on vrrp.50 for 192.168.20.3
May 12 14:18:46 Keepalived_vrrp[18764]: Sending gratuitous ARP on vrrp.50 for 192.168.20.3
May 12 14:18:46 Keepalived_vrrp[18764]: Sending gratuitous ARP on vrrp.50 for 192.168.20.3
May 12 14:18:51 Keepalived_vrrp[18764]: Sending gratuitous ARP on vrrp.50 for 192.168.20.3
May 12 14:18:51 Keepalived_vrrp[18764]: (VI_1) Sending/queueing gratuitous ARPs on vrrp.50 for 192.168.20.3
May 12 14:18:51 Keepalived_vrrp[18764]: Sending gratuitous ARP on vrrp.50 for 192.168.20.3
May 12 14:18:51 Keepalived_vrrp[18764]: Sending gratuitous ARP on vrrp.50 for 192.168.20.3
May 12 14:18:51 Keepalived_vrrp[18764]: Sending gratuitous ARP on vrrp.50 for 192.168.20.3
May 12 14:18:51 Keepalived_vrrp[18764]: Sending gratuitous ARP on vrrp.50 for 192.168.20.3

===HERE I RESET Devices B & C  and see error and log below===
May 12 14:20:22 Keepalived_vrrp[18764]: **Unknown VRID(2) received on interface(vrrp.50). ignoring...**
May 12 14:20:22 Keepalived_vrrp[18764]: **(VI_1) WARNING - equal priority advert received from remote host with our IP address.**
May 12 14:20:22 Keepalived_vrrp[18764]: (VI_1) WARNING - equal priority advert received from remote host with our IP address.
May 12 14:20:22 Keepalived_vrrp[18764]: (VI_1) WARNING - equal priority advert received from remote host with our IP address.
May 12 14:20:22 Keepalived_vrrp[18764]: (VI_1) WARNING - equal priority advert received from remote host with our IP address.
May 12 14:20:22 Keepalived_vrrp[18764]: (VI_1) WARNING - equal priority advert received from remote host with our IP address.
May 12 14:20:22 Keepalived_vrrp[18764]: (VI_1) WARNING - equal priority advert received from remote host with our IP address.
May 12 14:20:22 Keepalived_vrrp[18764]: (VI_1) WARNING - equal priority advert received from remote host with our IP address.
May 12 14:20:22 Keepalived_vrrp[18764]: (VI_1) WARNING - equal priority advert received from remote host with our IP address.

And many more lines with that Warning message.
Traffic is lost during that time.

What could be possibly the reason for such a behavior?
Couldn't find a useful information regarding these logs & error.

Thanks in advance.

**Comment by pqarmitage** [comment]
My guess is that this is something to do with your switch. If device B and/or C go down or are reset, keepalived on A should not know anything about it and so should not behave any differently from how it was before devices B and/or C go down.

There are two interesting aspects, as you have highlighted:
1. `Unknown VRID(2) received on interface(vrrp.50). ignoring...`. It would be useful to see the details of the VRRP packet and where it originated from.
2. `WARNING - equal priority advert received from remote host with our IP address.` My guess here is that the switch is looping the packet back to device A when it is the only device registered for listening on 224.0.0.18. Does your switch do IGMP snooping?

Running this using wireshark or tcpdump on device A would be interesting to actually see what is happening.

**Comment by rozmanro1** [comment]
> My guess is that this is something to do with your switch. If device B and/or C go down or are reset, keepalived on A should not know anything about it and so should not behave any differently from how it was before devices B and/or C go down.
> 
> There are two interesting aspects, as you have highlighted:
> 
> 1. `Unknown VRID(2) received on interface(vrrp.50). ignoring...`. It would be useful to see the details of the VRRP packet and where it originated from.
> 2. `WARNING - equal priority advert received from remote host with our IP address.` My guess here is that the switch is looping the packet back to device A when it is the only device registered for listening on 224.0.0.18. Does your switch do IGMP snooping?
> 
> Running this using wireshark or tcpdump on device A would be interesting to actually see what is happening.

It really seems like the switch is the problem.
In wireshark on Device_A we don't see anything that would confirm the looping.
no additional/weird packets.
But when tried to use other simple switch without the VLANS we don't see such behavior.

Thanks for your suggestion.
